src/sentry/taskworker/task.py

```diff
@@ -169,12 +177,41 @@ def create_activation(
                     f"The `{key}` header value is of type {type(value)}"
                 )
 
+        parameters_json = orjson.dumps({"args": args, "kwargs": kwargs})
+        if self.compression_type == CompressionType.ZSTD:
+            # TODO(taskworker): Nesting this conditional avoids django_db fixtures in tests.
```

Comment:
:+1: This also helps reduce overhead of checking options to only apply to tasks that are going to use compression.